src/libcollections/string.rs

```diff
@@ -701,6 +701,13 @@ impl String {
     /// Violating these may cause problems like corrupting the allocator's
     /// internal datastructures.
     ///
+    /// The ownership of `ptr` is effectively transferred to the
+    /// `String` and the memory referred to by the pointer will be
+    /// deallocated when the `String` is dropped. Ensure that
+    /// nothing else uses the pointer after calling this
+    /// function. Calling this function multiple times with the same
+    /// pointer is an error.
```

Comment:
Github overzealously hid the [unresolved previous conversation](https://github.com/rust-lang/rust/pull/34884#discussion_r71089040)